src/components/ValidateOrder.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { faX } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Input from './Input';
+import Button from './Button';
+import { useCreateOrderMutation } from '../states/api/apiSlice';
+import Loading from './Loading';
+import {
+  setOrderId,
+  updateModelVisility
+} from '../states/features/orders/orderSlice';
+
+const ValidateOrder = ({ visible, productName, productId, price }) => {
+  const [formData, setFormData] = useState({
+    productName,
+    productId,
+    price,
+    quantity: 1,
+    amount: price
+  });
+
+  const navigate = useNavigate();
+
+  const [
+    createOrder,
+    { data: orderData, error, isLoading, isError, isSuccess }
+  ] = useCreateOrderMutation();
+
+  const dispatch = useDispatch();
+
+  const handleSubmit = () => {
+    createOrder({
+      desiredQuantity: formData.quantity,
+      productId: formData.productId,
+      amount: formData.amount
+    });
+    if (isSuccess) {
+      dispatch(setOrderId(orderData.data.id));
+      navigate(`/order/${orderData.data.id}`)
+    }
+  };
```

Comment:
Same issue to the one of Frank. You can't capture isSuccess in the submission function. Try to use `useEffect` for that please. Work together with frank to resolve that.

Reply:
Please find the following workaround for your review:

```jsx
  const handleSubmit = () => {
    createOrder({
      desiredQuantity: formData.quantity,
      productId: formData.productId,
      amount: formData.amount
    });
  };

  useEffect(() => {
    if (isSuccess) {
      dispatch(setOrderId(orderData.data.id));
    }
  }, [orderData]);
```